app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitPromoCodesApiController.php

```diff
@@ -758,4 +759,60 @@ function ($attribute, $value, $fail) use ($filter) {
             return $this->ok();
         });
     }
+
+    /**
+     * @param $summit_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function sendSponsorPromoCodes($summit_id)
+    {
+        return $this->processRequest(function () use ($summit_id) {
+
+            if (!Request::isJson()) return $this->error400();
+
+            $summit = SummitFinderStrategyFactory::build($this->summit_repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit)) return $this->error404();
+
+            $payload = $this->getJsonPayload([
+                'email_flow_event' => 'required|string|in:' . join(',', [
+                        SponsorPromoCodeEmail::EVENT_SLUG,
+                    ]),
+                'promo_code_ids'          => 'sometimes|int_array',
+                'excluded_promo_code_ids' => 'sometimes|int_array',
+                'test_email_recipient'    => 'sometimes|email',
+                'outcome_email_recipient' => 'sometimes|email',
+            ]);
+
+            $filter = null;
+
+            if (Request::has('filter')) {
+                $filter = FilterParser::parse(Request::input('filter'), [
+                    'id'            => ['=='],
+                    'not_id'        => ['=='],
+                    'sponsor_company_name' => ['=@', '@@', '=='],
```

Comment:
its needs to match the filter described on the job its missing for example sponsor_id please double check it